org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/templates/XtextTemplateContext.java

```diff
@@ -88,37 +87,40 @@ protected TemplateTranslator createTemplateTranslator() {
 			int offset = getStart();
 			IRegion lineRegion = getDocument().getLineInformationOfOffset(offset);
 			String line = getDocument().get(lineRegion.getOffset(), lineRegion.getLength());
+			String lineDelimiter = getDocument().getLineDelimiter(0);
 			int i = 0;
-			while(i < line.length() && Character.isWhitespace(line.charAt(i))) {
+			while (i < line.length() && Character.isWhitespace(line.charAt(i))) {
 				i++;
 			}
 			if (i != 0)
-				return new IndentationAwareTemplateTranslator(line.substring(0, i));
-			return new TemplateTranslator();
-		} catch(BadLocationException ex) {
+				return new IndentationAwareTemplateTranslator(line.substring(0, i), lineDelimiter);
+			return new IndentationAwareTemplateTranslator("", lineDelimiter);
+		} catch (BadLocationException ex) {
 			return new TemplateTranslator();
 		}
 	}
-	
+
 	public static class IndentationAwareTemplateTranslator extends TemplateTranslator {
-	
+
 		private final String indentation;
+		private final String lineDelimiter;
 
-		public IndentationAwareTemplateTranslator(String indentation) {
+		public IndentationAwareTemplateTranslator(String indentation, String lineDelimiter) {
 			this.indentation = indentation;
+			this.lineDelimiter = lineDelimiter;
```

Comment:
if the given line delimiter is null, you can use the System's default.

Reply:
Done.